File: app/src/main/java/com/nelyan/fragments/ChatListFragment.java
package com.nelyan.fragments;

import android.app.Dialog;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.Spinner;
import android.widget.TextView;

import com.nelyan.AppUtils;
import com.nelyan.R;
import com.nelyan.adapter.ActivityListAdapter;
import com.nelyan.adapter.ChatListAdapter;
import com.nelyan.ui.Activity2Activity;
import com.nelyan.ui.ChildCareActivity;

import static com.nelyan.R.layout.customspinner;

public class ChatListFragment extends Fragment  implements AdapterView.OnItemSelectedListener{
    Context mContext;
    TextView tvFilter;
    ImageView ivButton,iv_map,ivBack;
    Spinner orderby;
    View v;
    Dialog dialog;
    ChatListAdapter chatListAdapter;
       RecyclerView recyclerview;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        v= inflater.inflate(R.layout.fragment_chat_list, container, false);
        mContext=getActivity();
        ivBack=v.findViewById(R.id.ivBack);
        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                FragmentManager fm = getActivity().getSupportFragmentManager();
                Fragment f = fm.findFragmentById(R.id.frame_container);
                fm.popBackStack();
            }
        });
     LinearLayout ll_1=v.findViewById(R.id.ll_1);
        ll_1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dailogLocation();
            }
        });
        orderby = v.findViewById(R.id.orderby);
        tvFilter=v.findViewById(R.id.tvFilter);
        tvFilter.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AppUtils.gotoFragment(mContext, new ChildCareFragment(), R.id.frame_container, false);
             //   AppUtils.gotoFragment(mContext, new HomeFragment(), R.id.container, false);
            }
        });
        iv_map=v.findViewById(R.id.iv_map);
        iv_map.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i= new Intent(getActivity(), ChildCareActivity.class);
                startActivity(i);
            }
        });
        recyclerview=v.findViewById(R.id.recyclerview);
        chatListAdapter = new ChatListAdapter(getActivity());
        recyclerview.setLayoutManager(new LinearLayoutManager(mContext));
        recyclerview.setAdapter(chatListAdapter) ;
        final String[] genderlist = new String[]{
                "",
                "Date Added",
                "Avilable Place",
                "Distance",
        };
        ArrayAdapter adapter = new ArrayAdapter(
                getActivity(), customspinner, genderlist);

// Setting Adapter to the Spinner
        orderby.setAdapter(adapter);

// Setting OnItemClickListener to the Spinner
        orderby.setOnItemSelectedListener(ChatListFragment.this);
        return v;
    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {

    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }
    public void dailogLocation(){

        dialog = new Dialog(mContext);
        dialog.getWindow().setBackgroundDrawableResource(android.R.color.transparent);
        dialog.setContentView(R.layout.alert_location);
        dialog.setCancelable(true);

        RelativeLayout rlYes,rlNo;
        rlNo= dialog.findViewById(R.id.rlNo);
        rlNo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //  mContext.startActivity(new Intent(mContext, HomeActivity.class));
                dialog.dismiss();
            }
        }); rlYes= dialog.findViewById(R.id.rlYes);
        rlYes.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                dialog.dismiss();
            }
        });
        dialog.show();
    }

}

File: app/src/main/java/com/nelyan_live/fragments/ChatListFragment.kt
/*
package com.nelyan_live.fragments

import android.app.Dialog
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import android.widget.AdapterView.OnItemSelectedListener
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.nelyan_live.AppUtils
import com.nelyan_live.R
import com.nelyan_live.R.layout
import com.nelyan_live.adapter.ChatListAdapter
import com.nelyan_live.ui.ChildCareActivity

class ChatListFragment : Fragment(), OnItemSelectedListener {
    lateinit  var mContext: Context
    var tvFilter: TextView? = null
    var ivButton: ImageView? = null
    var iv_map: ImageView? = null
    var ivBack: ImageView? = null
    var orderby: Spinner? = null
    lateinit  var v: View
    var dialog: Dialog? = null
    var chatListAdapter: ChatListAdapter? = null
    var recyclerview: RecyclerView? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        v = inflater.inflate(layout.fragment_chat_list, container, false)
        mContext = requireActivity()
        ivBack = v.findViewById(R.id.ivBack)
        ivBack!!.setOnClickListener(View.OnClickListener {
            val fm = requireActivity()!!.supportFragmentManager
            val f = fm.findFragmentById(R.id.frame_container)
            fm.popBackStack()
        })
        val ll_1 = v.findViewById<LinearLayout>(R.id.ll_1)
        ll_1.setOnClickListener { dailogLocation() }
        orderby = v.findViewById(R.id.orderby)
        tvFilter = v.findViewById(R.id.tvFilter)
        tvFilter!!.setOnClickListener(View.OnClickListener {
            AppUtils.gotoFragment(mContext, com.nelyan_live.fragments.ChildCareFragment(), R.id.frame_container, false)
            //   AppUtils.gotoFragment(mContext, new HomeFragment(), R.id.container, false);
        })
        iv_map = v.findViewById(R.id.iv_map)
        iv_map!!.setOnClickListener(View.OnClickListener {
            val i = Intent(activity, ChildCareActivity::class.java)
            startActivity(i)
        })
        recyclerview = v.findViewById(R.id.recyclerview)
        chatListAdapter = ChatListAdapter(requireActivity())
        recyclerview!!.setLayoutManager(LinearLayoutManager(mContext))
        recyclerview!!.setAdapter(chatListAdapter)
        val genderlist = arrayOf<String?>(
                "",
                "Date Added",
                "Avilable Place",
                "Distance")
        val adapter: ArrayAdapter<*> = ArrayAdapter<Any?>(
                requireActivity()!!, layout.customspinner, genderlist)

// Setting Adapter to the Spinner
        orderby!!.setAdapter(adapter)

// Setting OnItemClickListener to the Spinner
        orderby!!.setOnItemSelectedListener(this@ChatListFragment)
        return v
    }

    override fun onItemSelected(parent: AdapterView<*>?, view: View, position: Int, id: Long) {}
    override fun onNothingSelected(parent: AdapterView<*>?) {}
    fun dailogLocation() {
        dialog = Dialog(mContext!!)
        dialog!!.window!!.setBackgroundDrawableResource(android.R.color.transparent)
        dialog!!.setContentView(layout.alert_location)
        dialog!!.setCancelable(true)
        val rlYes: RelativeLayout
        val rlNo: RelativeLayout
        rlNo = dialog!!.findViewById(R.id.rlNo)
        rlNo.setOnClickListener { //  mContext.startActivity(new Intent(mContext, HomeActivity.class));
            dialog!!.dismiss()
        }
        rlYes = dialog!!.findViewById(R.id.rlYes)
        rlYes.setOnClickListener { dialog!!.dismiss() }
        dialog!!.show()
    }
}*/
